@here/harp-mapview/lib/AnimatedExtrusionHandler.ts

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (C) 2017-2019 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { TileKey } from "@here/harp-geoutils";
+import { DataSource } from "./DataSource";
+import { MapView } from "./MapView";
+import { Tile } from "./Tile";
+
+/**
+ * Animation states for extrusion effect
+ */
+export enum AnimatedExtrusionState {
+    None,
+    Started,
+    Playing,
+    Finished
+}
+
+/**
+ * Creates `AnimatedExtrusionHandler` that controls animated extrusion effect for
+ * extruded objects in [[Tile]] when they appear on the scene.
+ */
+export class AnimatedExtrusionHandler {
+    /**
+     * Animate the extrusion of the buildings if set to `true`.
+     */
+    static DEFAULT_ENABLED = true;
+    /**
+     * Duration of the building's extrusion in milliseconds
+     */
+    static DEFAULT_DURATION = 400;
```

Comment:
Sometimes the tiles pop in so fast on the corners of the screen that we don't see the animation. When trying locally I find that a duration of 1sec gives much more smoothness to the pan experience. That's personal though, as you prefer.

Reply:
replaced with 750ms as a compromise